api/util/util.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package util
+
+import (
+	"time"
+
+	"github.com/eclipse-kanto/update-manager/api/types"
+	"github.com/eclipse-kanto/update-manager/logger"
+)
+
+// ParseDuration converts the given string as time duration value.
+func ParseDuration(property, strValue string, defValue, emptyValue time.Duration) time.Duration {
+	if strValue == "" {
+		logger.Warn("Duration for property '%s' not set, using value %v", property, emptyValue)
+		return emptyValue
+	}
+	duration, err := time.ParseDuration(strValue)
+	if err != nil {
+		logger.Warn("Cannot parse duration for property '%s': %v, using default value %v", property, strValue, defValue)
+		return defValue
+	}
+	if duration < 0 {
+		logger.Warn("Negative duration for property '%s': %v, using default value %v", property, strValue, defValue)
+		return defValue
+	}
+	return duration
+}
+
+// FixIncompleteInconsistentStatus replaces StatusIncompleteInconsistent with StatusIncomplete, just a temporary workaround.
+func FixIncompleteInconsistentStatus(status types.StatusType) types.StatusType {
```

Comment:
Currently these two workaround functions are used only when sending feedback messages to the outside. 
And we need them because the cloud backend does not understand some of the status/actionStatus values yet. That's why we map them to known-to-backend values. 
Of course, this is not an elegant solution at all. There could be different solutions to this problem but this one is the easiest.
Anyway, the backend is in process of update to support the new status values. As soon as this is ready, the Fix methods shall be removed. I expect this to be done before the next Kanto release, so this is a well-known temporary workaround and it shall not be present in released version.